.github/workflows/IntegrationTesting.yaml

```
@@ -0,0 +1,188 @@
+        with:
+          name: sdk-flask-build-artifact
+          path: deploy.zip
+
+  deploy_WebApp:
+    name: Deploy Web Application
+    needs: build_WebApp
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest]
+        python-version: [3.8]
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - name: Checkout X-Ray SDK to get terraform source
+        uses: actions/checkout@v2
+
+      - name: Download WebApp with X-Ray SDK build artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: sdk-flask-build-artifact
+
+      - name: Copy deployment package to terraform directory
+        run: cp deploy.zip ./terraform
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
+
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v1
+
+      - name: Terraform Init
+        run: terraform init
+        working-directory: ./terraform
+
+      - name: Terraform Validate
+        run: terraform validate -no-color
+        working-directory: ./terraform
+
+      - name: Terraform Plan
+        run: terraform plan -var-file="fixtures.us-west-2.tfvars" -no-color
+        continue-on-error: true
+        working-directory: ./terraform
+
+      - name: Terraform Apply
+        run: terraform apply -var-file="fixtures.us-west-2.tfvars" -auto-approve
+        working-directory: ./terraform
+
+      - name: Upload terraform state files for destorying resources
+        uses: actions/upload-artifact@v2
+        with:
+          name: terraform-state-artifact
+          path: ./terraform
+
+  test_WebApp:
+    name: Test WebApp
+    needs: deploy_WebApp
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest]
+        python-version: [3.8]
+
+    steps:
+      - name: Pull in source code from TestRunner Github repository
+        uses: actions/checkout@v2
+        with:
+          repository: bhautikpip/TestRunner
+          ref: ${{ github.head_ref }}
+          token: ${{  secrets.PAT }}
+
+      - name: Setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
+
+      - name: Run Integration Test
+        run: |
+          pip3 install -r requirements.txt
+          python testrunner.py --t webapp --e http://eb-flask-app-env.us-west-2.elasticbeanstalk.com  --l python
+
+
+  cleanup:
+    name: Resource tear down
+    needs: test_WebApp
```

```suggestion
    needs: test_WebApp
    if: true
```

We want the cleanup to run regardless of the success or failure of the others.

---

Yes. I think this makes sense! very good find.